app/main.py

```
@@ -1,52 +1,38 @@
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.book_serializer import BookJSONSerializer, BookXMLSerializer
+from app.book_service import BookServiceConsole, BookServiceReverse
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+    service_map = {
+        "console": BookServiceConsole,
+        "reverse": BookServiceReverse,
+    }
+    serializer_map = {
+        "json": BookJSONSerializer,
+        "xml": BookXMLSerializer,
+    }
+
     for cmd, method_type in commands:
-        if cmd == "display":
-            book.display(method_type)
-        elif cmd == "print":
-            book.print_book(method_type)
-        elif cmd == "serialize":
-            return book.serialize(method_type)
+        if method_type in ["console", "reverse"]:
+            book_service = service_map.get(method_type)()
+
+            match cmd:
+                case "display":
+                    book_service.display(book)
+                case "print":
+                    book_service.print_book(book)
+                case _:
+                    raise ValueError(f"Unknown command {cmd}")
+
+        elif method_type in ["json", "xml"]:
+            serializer = serializer_map.get(method_type)()
+
+            match cmd:
+                case "serialize":
+                    return serializer.serialize(book)
+                case _:
+                    raise ValueError(f"Unknown command {cmd}")
```

The loop handles known `method_type` values for services and serializers, but if `method_type` is neither of those, the iteration is silently skipped and the function continues (ultimately returning `None` if nothing else returns). This can hide incorrect inputs. Consider adding an explicit `else:` that raises a clear error such as `ValueError(f"Unknown method_type {method_type}")` so callers get immediate, useful feedback. This change does not affect existing tests (they use valid method types) and improves correctness and debuggability.